test/e2e/components/vl-input-addon.js

```diff
@@ -1,6 +1,22 @@
 const { VlElement } = require('vl-ui-core').Test;
+const { By } = require('selenium-webdriver');
 
 class VlInputAddon extends VlElement {  
 }
 
-module.exports = VlInputAddon;
+class VlButtonInputAddon extends VlInputAddon {
+
+    async getIcon() {
+        const icon = await this.findElement(By.css(this.selector + ' [is="vl-icon"]'));
+        if (icon)  {
+            const { VlIcon } = require('vl-ui-icon').Test;
+            return new VlIcon(this.driver, icon);
+        } 
+    }
+ }
```

Comment:
zowel button als input-addon hebben geen 'source' dependency op icon, dus je kan je afvragen of het dan op zich wel koosjer is om een getIcon te voorzien daarop. we zijn nu al robuust tegen het feit dat je VlIcon niet moet hebben bij je dependencies als het er niet is, maar ge zou kunnen argumenteren dat deze methode er zelfs niet moet zijn en dat de afnemer zelf findElement maar moet doen. any thoughts @coemans ?

Reply:
Correct dat deze functie in principe niet nodig is. Ik denk dat de kans zeer groot is dat er een icon aanwezig zal zijn, dus lijkt het me wel een handige uitbreiding en zou ik ze behouden.